iceoryx_posh/include/iceoryx_posh/popo/condition.hpp

```diff
@@ -40,16 +50,17 @@ class Condition
   protected:
     friend class WaitSet;
     /// @brief User interface for specific attach of condition variable
-    virtual bool setConditionVariable(ConditionVariableData* const conditionVariableDataPtr) noexcept = 0;
+    virtual void setConditionVariable(ConditionVariableData* const conditionVariableDataPtr) noexcept = 0;
     /// @brief User interface for specific detach of condition variable
-    virtual bool unsetConditionVariable() noexcept = 0;
+    virtual void unsetConditionVariable() noexcept = 0;
 
     /// @brief Called by a WaitSet to announce the condition variable pointer that usually lives in shared memory
-    bool attachConditionVariable(ConditionVariableData* const conditionVariableDataPtr) noexcept;
+    void attachConditionVariable(WaitSet* const waitSet,
+                                 ConditionVariableData* const conditionVariableDataPtr) noexcept;
     /// @brief Called when removing the condition from a WaitSet
-    bool detachConditionVariable() noexcept;
+    void detachConditionVariable() noexcept;
 
-    std::atomic_bool m_conditionVariableAttached{false};
+    std::atomic<WaitSet*> m_waitSet{nullptr};
```

Comment:
Why this has to be atomic? Maybe it has to but we should know why

Reply:
If the WaitSet and Condition are not threadsafe then this does not need to be an atomic!
According to my understanding the normal usage is like this:

1. All in one thread
1.1.  Create WaitSet
1.2.  Create and attach Conditions
1.3.  Call WaitSet.wait() -> triggeredConditions
1.4.  Iterate through triggeredConditions and maybe start here some threads for callbacks.

Reply:
The question is how we would support attach/detach and wait in different threads. That's a setup I see. But then the question is if this would require the waitset ptr to be atomic. I think it is not used in the wait call